### ISSUE TITLE
Efficiency computation throws error for angular variable

### ISSUE THREAD

**Issue by ram1123** [vukasinmilosevic/Fiducial_XS#17]
For the five angular variables:

* cosTheta1
* cosTheta2
* Phi
* Phi1
* cosThetaStar

It gives error [1]. I didn't investigated why (possibly because some values get zero!!!)

@tjavaid 

[1]

```
Sample list: ['ggH_amcatnloFXFX_2018_slimmed', 'ZH_HToZZ_4LFilter_M125_2018_slimmed', 'testGGH_nnlops_GENonly_slimmed', 'ttH_HToZZ_4LFilter_M125_2018_slimmed', 'GluGluHToZZTo4L_M125_2018_slimmed', 'VBF_HToZZTo4L_M125_2018_slimmed', 'WH_HToZZTo4L_M125_2018_slimmed']
cutm4l_gen is.... (GENmass4l>105.0 && GENmass4l<140.0)
cutobs_gen is.... (abs(GENcosThetaStar)>=-3.14159 && abs(GENcosThetaStar)<-2.35619)
cutchan_gen is.... (abs(GENlep_id[GENlep_Hindex[0]])==13 && abs(GENlep_id[GENlep_Hindex[2]])==13)
acceptance    numerator======== 0.0
acceptance     denominator======== 361123.0
error checking processBin:  ggH_amcatnloFXFX_125_4mu_cosThetaStar_genbin0_recobin0
error checking histos processBin_reconoth4l integral:  0.0
error checking binfrac_wrongfrac:  0.0
Traceback (most recent call last):
  File "efficiencyFactors.py", line 708, in <module>
    geteffs(chan,List, m4l_bins, m4l_low, m4l_high, obs_reco, obs_gen, obs_bins, recobin, genbin)
  File "efficiencyFactors.py", line 331, in geteffs
    if (effrecotofid[processBin]*(1-effrecotofid[processBin])/Histos[processBin+"fid"].Integral()>0):
KeyError: 'ggH_amcatnloFXFX_125_4mu_cosThetaStar_genbin0_recobin0'
```

**Comment by tjavaid** [comment]
Hi @ram1123 , are you using efficiencyFactors.py from CMSSW_10_X_2018 branch or some other?

https://github.com/vukasinmilosevic/Fiducial_XS/blob/CMSSW_10_X_2018/FidXS_13TeV_2018_80X_preM19_newBinning_preApp/efficiencyFactors.py#L323-L338

In new binning we do not use absolute binning for cosThetaStar. 

https://github.com/vukasinmilosevic/Fiducial_XS/blob/CMSSW_10_X_2018/FidXS_13TeV_2018_80X_preM19_newBinning_preApp/doAllEffs_channels.sh#L48-L50

**Comment by vukasinmilosevic** [comment]
This is now implemented in your latest PR that was merged, can I close it?

**Comment by tjavaid** [comment]
yes, please.

**Comment by ram1123** [comment]
No, wait I see that there is still issue with the cos theta*

**Comment by vukasinmilosevic** [comment]
Ok, I saw that part being commented out from the variable list, that is why I asked. Will wait for an update.

**Comment by ram1123** [comment]
The main issue with the four variables:

- cosTheta1
- cosTheta2
- Phi
- Phi1

is the use of mod sign. When I removed the mod sign then the issue with the above mentioned obs. gone. But, the issue still there for costheta*

**Comment by tjavaid** [comment]
What binning are you using for cosThetaStar ? this one ?
https://github.com/vukasinmilosevic/Fiducial_XS/blob/CMSSW_10_X_2018/FidXS_13TeV_2018_80X_preM19_newBinning_preApp/doAllEffs_channels.sh#L48-L50
There was initially typo in binning of cosThetaStar which Alessandro provided as this variable was mixed with Phi variables. It should have same binning as cosTheta1, cosTheta2. Later he pointed to gDoc:
https://docs.google.com/spreadsheets/d/1KFUZu5Y_QXkuMHsxEQstRKC1HJF7qwVX4AYy5ve2gDs/edit#gid=927287546

In combination PR it is correct:
https://github.com/vukasinmilosevic/Fiducial_XS/blob/CMSSW_10_X_Combine/FidXS_13TeV_2018_80X_preM19_newBinning_preApp/doAllEffs_channels_2018.sh#L48-L50

**Comment by ram1123** [comment]
Hi @vukasinmilosevic ,

So, after updating  the costheta* binning

https://github.com/vukasinmilosevic/Fiducial_XS/blob/ae5a818c843e86a171fdfe95bbe47ded431163ab/Inputs/observables_list.yml#L54

to 

```
"|-1.0|-0.75|-0.5|-0.25|0.0|0.25|0.5|0.75|1.0|"
```

it worked.

Actually, I should notice this earlier. As costheta value can never exceed abs(1.0).

You can also include this in your next PR and close this issue.

**Comment by vukasinmilosevic** [comment]
Awesome, I will link the PR to this issue for book keeping and close it when merging the upcoming PR.

**Comment by vukasinmilosevic** [comment]
Fixed with #19
